akka-persistence-typed/src/main/scala/akka/persistence/typed/ReplicaId.scala

```diff
@@ -4,6 +4,15 @@
 
 package akka.persistence.typed
 
+object ReplicaId {
+
+  /**
+   * When migrating from non-replicated to replicated the ReplicaId of where the original entity
+   * was located should be empty.
+   */
+  val empty: ReplicaId = ReplicaId("")
+}
```

Comment:
Adding this companion makes mima complain:
```
[error] akka-persistence-typed: Failed binary compatibility check against com.typesafe.akka:akka-persistence-typed_2.13:2.6.12! Found 3 potential problems (filtered 18)
[error]  * static method andThen(scala.Function1)scala.Function1 in class akka.persistence.typed.ReplicaId does not have a correspondent in current version
[error]    filter with: ProblemFilters.exclude[DirectMissingMethodProblem]("akka.persistence.typed.ReplicaId.andThen")
[error]  * static method compose(scala.Function1)scala.Function1 in class akka.persistence.typed.ReplicaId does not have a correspondent in current version
[error]    filter with: ProblemFilters.exclude[DirectMissingMethodProblem]("akka.persistence.typed.ReplicaId.compose")
[error]  * the type hierarchy of object akka.persistence.typed.ReplicaId is different in current version. Missing types {scala.runtime.AbstractFunction1}
[error]    filter with: ProblemFilters.exclude[MissingTypesProblem]("akka.persistence.typed.ReplicaId$")
```

Strange, I thought that wouldn't be a problem. What shall I do? Add mima filter? Rename to `object EmptyReplicaId`?

Reply:
Huh, yeah maybe best to make a separate case object.

Reply:
huh, huh, and that doesn't work because can't extend final case class, and extending a case class is not good in the first place

Reply:
Added AbstractFunction1 to the companion.